src/SingleVault.sol

```diff
@@ -1,99 +1,64 @@
 // SPDX-License-Identifier: BSD-3-Clause
 pragma solidity ^0.8.24;
 
-import {
-    ERC4626Upgradeable,
-    ReentrancyGuardUpgradeable,
-    TimelockControllerUpgradeable,
-    IERC20,
-    IERC4626,
-    IStakeManager,
-    Math
-} from "src/Common.sol";
+import {ERC4626Upgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable, IERC20, Math} from "src/Common.sol";
 
-import {ISingleVault} from "src/ISingleVault.sol";
+import {ISingleVault} from "src/interface/ISingleVault.sol";
 
-contract SingleVault is ISingleVault, ERC4626Upgradeable, TimelockControllerUpgradeable, ReentrancyGuardUpgradeable {
-    using Math for uint256;
+/* ynETH Pre-Launch Vault */
 
-    bytes32 private constant ERC4626StorageLocation = 0x0773e532dfede91f04b12a73d3d2acd361424f41f76b4fb79f090161e36b4e00;
-    IStakeManager private constant stakeManager = IStakeManager(0x1adB950d8bB3dA4bE104211D5AB038628e477fE6);
+contract SingleVault is ISingleVault, ERC4626Upgradeable, AccessControlUpgradeable, ReentrancyGuardUpgradeable {
```

Comment:
the unstated assumption for this implementation is that whatever we pick for 

asset() is 1:1 to ETH, which is fine

Reply:
@danoctavian so you're saying the underlying asset of the ETH vault is wstETH?

Reply:
no, i's not, it will be WETH

the implementation of receive() actually clears the fact that asset() must be WETH so it's clear asset() = WETH